cmake/kconfig.cmake

```diff
@@ -104,6 +104,38 @@ if(EXTRA_KCONFIG_OPTIONS)
     )
 endif()
 
+# Set overlay config files list using a list of directories where the
+# overlay files could be found.
+if(DEFINED ENV{OVERLAY_CONFIG_DIRS})
+  set(OVERLAY_CONFIG_DIRS $ENV{OVERLAY_CONFIG_DIRS})
+
+  # We add current directory first so that it is seached first
+  set(overlay_config_dir_list ".:${OVERLAY_CONFIG_DIRS}")
+  string(REPLACE ":" ";" overlay_config_dirs "${overlay_config_dir_list}")
+
+  foreach(f IN ITEMS ${OVERLAY_CONFIG_AS_LIST})
+    if(IS_ABSOLUTE ${f})
+      list(APPEND overlay_config_files ${f})
+    else()
+      set(found 0)
+      foreach(d ${overlay_config_dirs})
```

Comment:
The word 'overlay' is DT terminology.

In Kconfig, Kconfig fragments are merged together, not overlayed.

I know some samples are calling their fragments overlays, but they shouldn't do so,
and up until now we haven't been using the word 'overlay' in official documentation,
nor should we start to do so now.

So please use a different word.